services/state/roster/roster_state.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+package com.hedera.hapi.node.state.roster;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "basic_types.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.roster">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * The current state of platform rosters.<br/>
+ * This message stores a roster data for the platform in network state.
+ *
+ * The roster state SHALL encapsulate the incoming candidate roster's hash,<br/>
+ * and a list of pairs of round number and active roster hash.<br/>
+ * This data SHALL be used to track round numbers and the rosters used in determining the consensus.<br/>
+ */
+message RosterState {
+
+  /**
+   * The SHA-384 hash of a candidate roster.
+   * <p>
+   * This is the hash of the roster that is currently being considered
+   * for adoption.
+   * A Node SHALL NOT, ever, have more than one candidate roster
+   * at the same time.
+   */
+  byte candidate_roster_hash = 1;
+
+  /**
+   * A list of round numbers and roster hashes.
+   * The round number indicates the round in which the corresponding roster became active
+   * <p>
+   * This list SHALL be ordered by round numbers.<br/>
```

Comment:
In ascending or descending order? It might be worth specifying this here.

Reply:
I think specifying ascending here is kind of tautological. There's nowhere in the hashgraph logic where anything related to round numbers can possibly be descending. I think it's fine to leave it.

Reply:
Why? One could imagine that an element at index zero could return the "active" or "latest" entry, which would imply a descending order. And this may make sense as fetching the element at index zero is the fastest operation because it doesn't require the knowledge about the size of your list.

So the actual order is not very clear from the specification. I suggest to specify it explicitly to avoid any confusion or misinterpretation.

Reply:
applied your suggestion.